packages/manager/modules/pci/src/projects/project/kubernetes/components/autoscaling/autoscaling.controller.js

```diff
@@ -0,0 +1,142 @@
+  constructor(coreConfig) {
+    this.ovhSubsidiary = coreConfig.getUser().ovhSubsidiary;
+    this.autoscalingInfoLink = this.getAutoscalingInfoLink();
+  }
+
+  $onInit() {
+    this.setLowestHighestLimit();
+    this.model.isValidScale = AutoscalingController.isValidKubeScale(
+      this.model.autoscale,
+      this.model.nodes.lowest.value,
+      this.model.nodes.desired.value,
+      this.model.nodes.highest.value,
+    );
+  }
+
+  getAutoscalingInfoLink() {
+    return AUTOSCALING_LINK[this.ovhSubsidiary] || AUTOSCALING_LINK.DEFAULT;
+  }
+
+  resetScaleValuesToDefault() {
+    const { lowest, highest } = this.model.nodes;
+    lowest.min = SCALE_DEFAULT_VALUES.LOWEST_MIN_VALUE;
+    lowest.value = SCALE_DEFAULT_VALUES.LOWEST_MIN_VALUE;
+
+    highest.value = SCALE_DEFAULT_VALUES.HIGHEST_VALUE;
+    highest.max = SCALE_DEFAULT_VALUES.HIGHEST_MAX_VALUE;
+  }
+
+  setLowestHighestLimit() {
+    const { lowest, desired, highest } = this.model.nodes;
+    lowest.max = desired.value;
+    highest.min = desired.value;
+  }
+
+  onAutoscaleChanged(autoscale) {
+    if (!autoscale.status) {
+      this.resetScaleValuesToDefault();
+    } else {
+      this.setLowestHighestLimit();
+    }
+
+    AutoscalingController.triggerHandlerFunction(
+      this.onAutoscalePoolChanged,
+      autoscale,
+    );
+  }
+
+  onLowestValueChanged(min) {
+    const { lowest, desired, highest } = this.model.nodes;
+
+    highest.min = min.value;
+
+    if (min.value >= desired.value) {
+      lowest.max = min.value;
+    }
+
+    this.model.isValidScale = AutoscalingController.isValidKubeScale(
+      this.model.autoscale,
+      min.value,
+      desired.value,
+      highest.value,
+    );
+
+    AutoscalingController.triggerHandlerFunction(
+      this.onLowestPoolValueChanged,
+      min,
+    );
+  }
+
+  onDesiredValueChanged(need) {
+    const { lowest, highest } = this.model.nodes;
+
+    lowest.max = need.value;
+    highest.min = need.value;
+
+    if (need.value <= lowest.value) {
+      lowest.value = need.value;
+    }
+
+    if (need.value >= highest.value) {
+      highest.value = need.value;
+    }
+
+    this.model.isValidScale = AutoscalingController.isValidKubeScale(
+      this.model.autoscale,
+      lowest.value,
+      need.value,
+      highest.value,
+    );
+
+    AutoscalingController.triggerHandlerFunction(
+      this.onDesiredPoolValueChanged,
+      need,
+    );
+  }
+
+  onHighestValueChanged(max) {
+    const { lowest, desired, highest } = this.model.nodes;
+
+    lowest.max = max.value;
+
+    if (max.value <= desired.value) {
+      highest.min = max.value;
+    }
+
+    this.model.isValidScale = AutoscalingController.isValidKubeScale(
+      this.model.autoscale,
+      lowest.value,
+      desired.value,
+      max.value,
+    );
+
+    AutoscalingController.triggerHandlerFunction(
+      this.onHighestPoolValueChanged,
+      max,
+    );
+  }
+
+  static triggerHandlerFunction(fn, data) {
+    if (isFunction(fn)) fn(data);
```

Comment:
```suggestion
    if (typeof fn === 'function') {
      fn(data)
    };
```
Or, to keep one line you could also, and avoid with no brackets: (nitpick)
```suggestion
    typeof fn === 'function' && fn(data);
```

Reply:
Thanks for suggestion.
Initially i was wrote it like the second suggestion, but finally i preferred to use Lodash function (elegant).

Reply:
I agree `isFunction` lodash method is more elegant. But in fact you're importing a dependency here, and so even if it small you're adding weight to your code. And IMO it is no useful as it is just for cosmetic details

Reply:
Right, fixed.